### PR TITLE
feat: voices treasury space

### LINKED ISSUE #719: 🛠️ [TASK] : Treasury Space MVE1 Mocked UI
### Summary

Mocked Treasury Space Screen

### Description

Development of Treasury Space screen 

- [Figma](https://www.figma.com/design/80aDAQgoD0v06DO2Sb76ok/Catalyst--Minimal-Valuable-Experiments-(MVE's)?node-id=5152-60122&m=dev)

Part of #726

### PR DESCRIPTION
**Pull request by damian-molinski**
# Description

Voices Treasury UI

## Related Issue(s)

Closes #719 

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/lib/pages/treasury/campaign_segment_controller.dart [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import 'package:catalyst_voices/widgets/menu/voices_node_menu.dart';
+import 'package:flutter/material.dart';
+
+typedef ControllerBuilder = CampaignController Function(
```
Comment (dtscalac) [inline catalyst_voices/lib/pages/treasury/campaign_segment_controller.dart:4]: CampaignControllerBuilder? The reason is ControllerBuilder is quite generic and in flutter we have plenty of different controllers

File: catalyst_voices/lib/widgets/menu/voices_node_menu.dart [hunk 2]
```diff
@@ -51,64 +57,79 @@ final class VoicesNodeMenuData extends Equatable {
...
 
+  @override
+  State<VoicesNodeMenu> createState() => _VoicesNodeMenuState();
+}
+
+class _VoicesNodeMenuState extends State<VoicesNodeMenu> {
+  VoicesNodeMenuController? _controller;
+
+  VoicesNodeMenuController get _effectiveController =>
+      widget.controller ?? (_controller ??= VoicesNodeMenuController());
```
[the remark this answers is not in the document]
Comment (damian-molinski) [reply catalyst_voices/lib/widgets/menu/voices_node_menu.dart:102]: Actually this is quite common pattern. For example `TextField` uses it for `focusNode` (https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/text_field.dart#L986)

File: catalyst_voices/lib/pages/treasury/campaign_segment_controller.dart (hunk 1 above)
Comment (damian-molinski) [reply catalyst_voices/lib/pages/treasury/campaign_segment_controller.dart:4]: Agree

File: catalyst_voices/lib/widgets/menu/voices_node_menu.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (dtscalac) [reply catalyst_voices/lib/widgets/menu/voices_node_menu.dart:102]: True, but I think TextEditingController example is more relevant, they decided to implement didUpdateWidget correctly: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/text_field.dart#L1127.

I don't think in our case it's that important to have it, some things even if they are kept in memory a little bit longer are not huge issues.

**Review by dtscalac — APPROVED** [review]

**Pull request merged**